toolchain/sem_ir/ids.h

```diff
@@ -472,6 +475,64 @@ struct ElementIndex : public IndexBase, public Printable<ElementIndex> {
   }
 };
 
+// The ID of an ImportIRInst.
+struct ImportIRInstId : public IdBase, public Printable<InstId> {
+  using ValueType = ImportIRInst;
+
+  using IdBase::IdBase;
+};
+
+// A SemIR location used exclusively for diagnostic locations. If positive, it
+// corresponds to a Parse::NodeId in the current IR. If valid and negative, it
+// corresponds to an ImportIRInstId.
```

Comment:
What about zero?

Also, the code isn't implemented in terms of "positive" and "negative", it's implemented in terms of comparisons with `InvalidIndex`. Might it be clearer to phrase the comment in those terms as well? Alternatively, maybe the mapping from integers to sub-kinds of location should be treated as an implementation detail, and not specified here at all?

Reply:
Adjusted comments. I think the implementation is nuanced and should be commented somewhere; here seems best.